x-pack/plugins/observability_solution/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -51,21 +50,29 @@ export async function createAnomalyDetectionJobs({
 
   return withApmSpan('create_anomaly_detection_jobs', async () => {
     logger.info(`Creating ML anomaly detection jobs for environments: [${uniqueMlJobEnvs}].`);
-
     const apmMetricIndex = indices.metric;
     const responses = [];
     const failedJobs = [];
+
+    if (!esCapabilities.serverless) {
+      // Waiting for the index is not enabled in serverless, this could potentially cause
+      // problems when creating jobs in parallel
+      try {
+        await waitForIndexStatus(esClient, apmMetricIndex, 'yellow');
```

Comment:
* I have placed the "wait for yellow" BEFORE attempting to `createAnomalyDetectionJob`.
* I have placed it outside of the loop (aka only one `esClient.cluster.health` call before loop-creating anomaly detection jobs).
* I am using the `apmMetricIndex` param instead of the previously hardcoded `.ml-*`.

@yngrdyn @awahab07 do you confirm you are okay with this changes?